communication/src/TlsConnectionManager.cpp

```diff
@@ -341,6 +354,17 @@ void ConnectionManager::connect() {
   }
 }
 
+void ConnectionManager::handleConnectTimeout(NodeNum i, const std::error_code& error_code) {
+  asio::steady_timer& deadline_timer = connecting_.at(i).second;
+  if (deadline_timer.expiry() <= asio::steady_timer::clock_type::now()) {
+    LOG_WARN(logger_, "Timeout during connecting to node : " << i << ": " << error_code.message());
+    connecting_.at(i).first.close();
+    status_->num_connecting = connecting_.size();
```

Comment:
As we talked yesterday, status_->num_connecting = connecting_.size(); should be called after erase